postgres/datadog_checks/postgres/relationsmanager.py

```diff
@@ -107,27 +107,54 @@
 
 # The catalog pg_class catalogs tables and most everything else that has columns or is otherwise similar to a table.
 # For this integration we are restricting the query to ordinary tables.
-SIZE_METRICS = {
-    'descriptors': [('nspname', 'schema'), ('relname', 'table')],
-    'metrics': {
-        'pg_table_size(C.oid) as table_size': ('postgresql.table_size', AgentCheck.gauge),
-        'pg_indexes_size(C.oid) as index_size': ('postgresql.index_size', AgentCheck.gauge),
-        'pg_total_relation_size(C.oid) as total_size': ('postgresql.total_size', AgentCheck.gauge),
-    },
-    'relation': True,
+#
+# Table size: We want to get toast size separated from table size.
+# We can't use pg_total_relation_size which include both toast and table size.
+# We use pg_relation_size which only get the size of the main fork.
+#
+# Calls to pg_indexes_size and pg_relation_size needs to get a lock on the relation.
+# While cheap, it's cheaper to avoid calling it if we know there's no index. Same for toast size.
+#
+# Previous version filtered on nspname !~ '^pg_toast'. Since pg_toast namespace only
+# contains index and toast table, the filter was redundant with relkind = 'r'
+QUERY_PG_CLASS = {
+    'name': 'pg_class',
     'query': """
-SELECT
-  N.nspname,
-  relname,
-  {metrics_columns}
-FROM pg_class C
-LEFT JOIN pg_namespace N ON (N.oid = C.relnamespace)
-WHERE nspname NOT IN ('pg_catalog', 'information_schema') AND
-  nspname !~ '^pg_toast' AND
-  relkind = 'r' AND
-  {relations}""",
+SELECT current_database(),
+       s.schemaname, s.table, s.partition_of,
+       s.relpages, s.reltuples, s.relallvisible,
+       s.table_size, s.index_size, s.toast_size, s.table_size + s.index_size + s.toast_size
```

Comment:
> Change metric: postgresql.table_size now only reports the size of the main table (removing toast size, fsm and vm)
> New metric: postgresql.toast_size reports the size of the associated toast table

@bonnefoa is this desirable for customers? I certainly support the new metrics with additional granularity, but if you can get derive new metrics from existing metrics (like your new/changed `postgresql.table_size` = current `postgresql.table_size` - `postgresql.toast_size`), I'm hesitant to change the definition and value of an existing metric.

My concern is that there are 20-ish customers who have monitors on the `postgresql.table_size` metric and changing the definition may be undesirable if it is inconsistent with the common understanding of the colloquial "size of a table," which I would expect most customers to consider "size of data stored as a result of this table" (i.e. inclusive of TOAST).

I found this SO post: https://stackoverflow.com/a/70397779

![image](https://user-images.githubusercontent.com/717862/236488545-9f28edd2-5d1c-4435-bc53-6d2fd027377e.png)

Reply:
Confirmed with @bonnefoa on slack that this thread was resolved. Just wanted to be sure this was settled before we go forward with this.